File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

class Main{
    static int n, m, v;
    static ArrayList<ArrayList<Integer>> graph = new ArrayList<>();
    static int[] visited;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String input=br.readLine();
        StringTokenizer st = new StringTokenizer(input);
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());
        for (int i = 0; i <= n; i++) {
            graph.add(new ArrayList<Integer>());
        }
        for (int i = 0; i < m; i++) {
            String input2 = br.readLine();
            st = new StringTokenizer(input2);
            int a, b;
            a = Integer.parseInt(st.nextToken());
            b = Integer.parseInt(st.nextToken());
            graph.get(a).add(b);
            graph.get(b).add(a);
        }
        for (int i = 0; i <=n ; i++) {
            Collections.sort(graph.get(i));
        }
        visited=new int[n+1];
        dfs(v);
        System.out.println();
        visited=new int[n+1];
        bfs(v);
        System.out.println();

    }

    private static void bfs(int v) {
        Queue<Integer> q = new LinkedList<>();
        q.offer(v);
        visited[v]=1;
        while (!q.isEmpty()) {
            int now = q.poll();
            System.out.print(now+" ");
            for (int i = 0; i < graph.get(now).size(); i++) {
                if (visited[graph.get(now).get(i)] == 0) {
                    q.offer(graph.get(now).get(i));
                    visited[graph.get(now).get(i)]=1;
                }
            }
        }
    }

    private static void dfs(int v) {
        visited[v]=1;
        System.out.print(v+" ");
        for (int i = 0; i < graph.get(v).size(); i++) {
            if(visited[graph.get(v).get(i)]==0){
                dfs(graph.get(v).get(i));
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cs
using System;
using System.Collections.Generic;
using System.IO;

class Program
{
    static int n, m, v;
    private static List<List<int>> graph = new List<List<int>>();
    private static bool[] visited;

    public static void Main(string[] args)
    {
        string[] tmp = Console.ReadLine().Split(" ");
        n = int.Parse(tmp[0]);
        m = int.Parse(tmp[1]);
        v = int.Parse(tmp[2]);

        for (int i = 0; i <= n; i++)
        {
            graph.Add(new List<int>());
        }

        for (int i = 0; i < m; i++)
        {
            var strings = Console.ReadLine().Split(" ");
            int a = int.Parse(strings[0]);
            int b = int.Parse(strings[1]);

            graph[a].Add(b);
            graph[b].Add(a);
        }

        for (int i = 1; i <= n; i++)
        {
            graph[i].Sort();
        }

        visited = new bool[n + 1];
        dfs(v);
        Console.WriteLine();
        visited = new bool[n + 1];
        bfs(v);
    }

    static void dfs(int now)
    {
        Console.Write(now + " ");
        if (!visited[now])
        {
            visited[now] = true;
        }

        for (int i = 0; i < graph[now].Count; i++)
        {
            int next = graph[now][i];
            if (visited[next]) continue;

            visited[next] = true;
            dfs(next);
        }
    }

    static void bfs(int now)
    {
        Queue<int> q = new Queue<int>();
        q.Enqueue(now);
        visited[now] = true;
        while (q.Count > 0)
        {
            int current = q.Dequeue();
            Console.Write(current + " ");
            for (int i = 0; i < graph[current].Count; i++)
            {
                int next = graph[current][i];
                if (visited[next]) continue;

                visited[next] = true;
                q.Enqueue(next);
            }
        }
    }
}